### PR TITLE
Add negative binom to pygam

### LINKED ISSUE #302: Add negative binomial for GAM
- [x] Additional function parameters, change functionality or change defaults?
- [ ] New estimator in `cr.tl.estimators`?
- [ ] New kernel in `cr.tl.kernels`?
- [ ] New gene trend model in `cr.ul.models`?
- [ ] New plotting function in `cr.pl`?
- [ ] Other?

`pygam` is fairly flexible so it shouldn't be a problem.

### PR DESCRIPTION
**Pull request by michalk8**
**IMPORTANT: Please search among the [Pull request](../pulls) before creating one.**

## Title

Implements negative binomial for pygam.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] New feature (non-breaking change which adds functionality)

## Description

See above.

## How has this been tested?

It hasn't. Will be in the mega test PR before official release (promise).

## Closes

closes #302

### CODE REVIEW

**Comment by Marius1311** [comment]
Does this slow down the fit significantly? Something tells me that for Gaussian noise, this is much faster to fit than for NB.

**Comment by michalk8** [comment]
@Marius1311 when you find the time, could you please check the math? I've been using this as an resource https://courses.ms.ut.ee/MTMS.01.011/2018_spring/uploads/Main/GLM_slides_9_NB.pdf + https://support.sas.com/documentation/cdl/en/statug/63033/HTML/default/statug_genmod_sect030.htm + https://en.wikipedia.org/wiki/Negative_binomial_distribution, but I may have become more confused.

I will check out the performance later.

**Comment by Marius1311** [comment]
Hi Mike, we know by now that droplet based single cell data is not zero inflated, so no need to include zinb. Which parts of the maths would you like me to check? Can you be more specific? i.e. are you worried about the distribution, about the optimization, about the linear model, etc.  We can talk about this tomorrow in our call.

**Comment by michalk8** [comment]
I'm worried about the distribution, since that's the only thing I've touched. I am including comments in the code. Let's talk about it tomorrow.

File: cellrank/ul/models/_models.py [hunk 1]
```diff
@@ -1501,3 +1533,118 @@ def _maybe_import_r_lib(
...
+        %(nb_y)s
+        %(nb_mu)s
+        scaled
+            Whether to divide the deviance by the distribution scaled.
+
+        Returns
+        -------
+           The deviances, array of shape `(n,)`.
+        """
+        dev = 2 * (
```
Comment (michalk8) [inline cellrank/ul/models/_models.py:1620]: ping @Marius1311 - math check

**Comment by Marius1311** [comment]
Can you check against scipy `scipy.stats.nbinom`? i.e. just sample repeatedly from both and see whether they converge.

**Comment by michalk8** [comment]
This is by now fairly outdated, closing in favor of #377.

**Pull request closed without merging**